test/unit/spec/narrative_core/upload/importSetup-spec.js

```diff
@@ -0,0 +1,190 @@
+            TestUtil.clearRuntime();
+        });
+
+        it('should have expected functions', () => {
+            ['setupImportCells', 'setupWebUploadCell'].forEach((fn) => {
+                expect(ImportSetup[fn]).toEqual(jasmine.any(Function));
+            });
+        });
+
+        describe('setupImportCells tests', () => {
+            const TYPE = {
+                FASTQ: 'fastq_reads_interleaved', // bulk
+                SRA: 'sra_reads', // bulk
+                MATRIX: 'expression_matrix', // not bulk
+                MEDIA: 'media', // not bulk
+            };
+
+            const appIds = Object.values(TYPE).reduce((acc, appType) => {
+                acc[appType] = uploaders.app_info[appType].app_id;
+                return acc;
+            }, {});
+
+            const BULK = new Set([TYPE.FASTQ, TYPE.SRA]);
+            const fileCounts = [1, 2, 5];
+
+            /**
+             * Builds the fileInfo array with the given number of files of each data type.
+             * Returns an object with:
+             * {
+             *   fileInfo: array of file information (inputs to setupImportCells)
+             *   biCell: boolean - if true, then a single BI cell should get made
+             * }
+             */
+            function setupImportCellsInputs(numFiles, dataTypes) {
+                let biCell = false;
+                const fileInfo = [],
+                    bulkFiles = new Set(),
+                    nonBulkFiles = new Set();
+                for (let i = 0; i < dataTypes.length; i++) {
+                    for (let j = 0; j < numFiles; j++) {
+                        const fileName = `file${i * numFiles + j + 1}`;
+                        fileInfo.push({
+                            name: fileName,
+                            type: dataTypes[i],
+                        });
+                        if (BULK.has(dataTypes[i])) {
+                            biCell = true;
+                            bulkFiles.add(fileName);
+                        } else {
+                            nonBulkFiles.add(fileName);
+                        }
+                    }
+                }
+                return {
+                    fileInfo,
+                    biCell,
+                    bulkFiles,
+                    nonBulkFiles,
+                };
+            }
+
+            // multiple file types, multiple of each
+            // not gonna make a permuter because that's just overkill.
+            const typeCombos = [
+                ['FASTQ', 'SRA'],
+                ['FASTQ', 'MATRIX'],
+                ['FASTQ', 'SRA', 'MATRIX'],
+                ['FASTQ', 'MATRIX', 'MEDIA'],
+                ['FASTQ', 'SRA', 'MATRIX', 'MEDIA'],
+            ].concat(Object.keys(TYPE).map((singleType) => [singleType]));
+            fileCounts.forEach((numFiles) => {
+                typeCombos.forEach((typeCombo) => {
+                    const types = typeCombo.map((type) => TYPE[type]);
```

Comment:
I may have gotten a little too carried away with being clever here.

The main tests that happen are adding lots of files and type permutations, sending those to `setupImportCells` and tracking how the various cells get created. This includes some combinations of types, along with each individual type on its own, with a number of files ranging 1, 2, 5 of each type. It then combines each of those to make unique files for all types, runs the function, and sees that the cells get created with expected values (bulk import cells only use configured bulk apps, singleton import cells only use the right files, etc.)

There's probably more that could be done here, but that will come with better error coverage. Right now, this is all happy path-ish.